model/project_parser.go

```diff
@@ -581,15 +581,20 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 			return nil, errors.Wrapf(err, LoadProjectError)
 		}
 		opts.RemotePath = path.FileName
-		// read from the patch diff if a change has been made in any of the include files
-		if opts.ReadFileFrom == ReadFromPatch || opts.ReadFileFrom == ReadFromPatchDiff {
-			if opts.PatchOpts.patch != nil && opts.PatchOpts.patch.ConfigChanged(path.FileName) {
-				opts.ReadFileFrom = ReadFromPatchDiff
-			} else {
-				opts.ReadFileFrom = ReadFromPatch
+		var yaml []byte
+		if path.Module != "" {
+			yaml, err = retrieveFileForModule(ctx, *opts, intermediateProject.Modules, path.Module)
+		} else {
+			// read from the patch diff if a change has been made in any of the include files
+			if opts.ReadFileFrom == ReadFromPatch || opts.ReadFileFrom == ReadFromPatchDiff {
```

Comment:
I think this should still be done outside of the if/else, bc retrieveFileForModule will also eventually need to consider the differences here)

Reply:
done